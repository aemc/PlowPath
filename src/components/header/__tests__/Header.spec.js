import React from "react";
import Component from "../Header";
import { shallow } from "enzyme";
import PropTypes from "prop-types";

const defaultProps = {
  classes: {},
  open: false,
  setOpen: jest.fn()
};

describe("Header", () => {
  it("renders", () => {
    const wrapper = shallow(<Component {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
