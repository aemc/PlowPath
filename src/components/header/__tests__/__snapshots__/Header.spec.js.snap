// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Header)
    classes={Object {}}
    open={[Function]}
    setOpen={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "appBar": "Header-appBar-2",
        "appBarShift": "Header-appBarShift-3",
        "content": "Header-content-10",
        "drawer": "Header-drawer-6",
        "drawerClose": "Header-drawerClose-8",
        "drawerOpen": "Header-drawerOpen-7",
        "hide": "Header-hide-5",
        "menuButton": "Header-menuButton-4",
        "root": "Header-root-1",
        "toolbar": "Header-toolbar-9",
      },
      "open": [Function],
      "setOpen": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "appBar": "Header-appBar-2",
          "appBarShift": "Header-appBarShift-3",
          "content": "Header-content-10",
          "drawer": "Header-drawer-6",
          "drawerClose": "Header-drawerClose-8",
          "drawerOpen": "Header-drawerOpen-7",
          "hide": "Header-hide-5",
          "menuButton": "Header-menuButton-4",
          "root": "Header-root-1",
          "toolbar": "Header-toolbar-9",
        },
        "open": [Function],
        "setOpen": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
